### PR TITLE
Add failure actions FFI

### PR DESCRIPTION
**Pull request by pronebird**
This PR adds missing FFI for managing the failure actions for windows services:

Relevant documentation:

- https://docs.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-_service_failure_actionsa
- https://docs.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-sc_action
- https://docs.microsoft.com/en-us/windows/win32/api/winsvc/ns-winsvc-_service_failure_actions_flag

### CODE REVIEW

**Comment by pronebird** [comment]
Bump. Any chance to get this merged, all checks passed and we use it in production via public fork...

**Comment by pronebird** [comment]
@retep998 I hate pinging you directly, but it's been a month. Any way of merging this so we could switch back our origins to winapi-rs instead of our fork? .... thanks

File: src/um/winsvc.rs [hunk 1]
```diff
@@ -185,6 +185,28 @@ DEFINE_GUID!{NAMED_PIPE_EVENT_GUID,
...
+    dwResetPeriod: DWORD,
+    lpRebootMsg: LPWSTR,
+    lpCommand: LPWSTR,
+    cActions: DWORD,
+    lpsaActions: LPSC_ACTION,
+}}
+pub type LPSERVICE_FAILURE_ACTIONSW = *mut SERVICE_FAILURE_ACTIONSW;
+STRUCT!{struct SERVICE_FAILURE_ACTIONS_FLAG {
+    fFailureActionsOnNonCrashFailures: BOOL,
+}}
```
Comment (Connicpu) [inline src/um/winsvc.rs:209]: 👍 definition matches the headers

**Review by Connicpu — APPROVED** [review]
All of it checks out, pretty small addition

**Review by retep998 — APPROVED** [review]

**Pull request merged**